python/lsst/analysis/tools/actions/vector/calcRhoStatistics.py

```diff
@@ -0,0 +1,649 @@
+
+    colXy = Field[str](doc="The column name to get the xy shape component from.", default="{band}_ixy")
+
+    colPsfXx = Field[str](
+        doc="The column name to get the PSF xx shape component from.", default="{band}_ixxPSF"
+    )
+
+    colPsfYy = Field[str](
+        doc="The column name to get the PSF yy shape component from.", default="{band}_iyyPSF"
+    )
+
+    colPsfXy = Field[str](
+        doc="The column name to get the PSF xy shape component from.", default="{band}_ixyPSF"
+    )
+
+    ellipticityType = ChoiceField[str](
+        doc="The type of ellipticity to calculate",
+        allowed={
+            "chi": "Distortion, defined as (Ixx - Iyy)/(Ixx + Iyy)",
+            "epsilon": ("Shear, defined as (Ixx - Iyy)/" "(Ixx + Iyy + 2*sqrt(Ixx*Iyy - Ixy**2))"),
+        },
+        default="chi",
+    )
+
+    sizeType = ChoiceField[str](
+        doc="The type of size to calculate",
+        default="trace",
+        allowed={
+            "trace": "trace radius",
+            "determinant": "determinant radius",
+        },
+    )
+
+    treecorr = ConfigField[BinnedCorr2Config](
+        doc="TreeCorr configuration",
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.treecorr = BinnedCorr2Config()
+        self.treecorr.sep_units = "arcmin"
+        self.treecorr.max_sep = 100.0
+        self.treecorr.metric = "Arc"
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return (
+            (self.colRa, Vector),
+            (self.colDec, Vector),
+            (self.colXx, Vector),
+            (self.colYy, Vector),
+            (self.colXy, Vector),
+            (self.colPsfXx, Vector),
+            (self.colPsfYy, Vector),
+            (self.colPsfXy, Vector),
+        )
+
+    def __call__(self, data: KeyedData, **kwargs) -> KeyedData:
+        calcEMeas = CalcE(
+            colXx=self.colXx,
+            colYy=self.colYy,
+            colXy=self.colXy,
+            ellipticityType=self.ellipticityType,
+        )
+        calcEpsf = CalcE(
+            colXx=self.colPsfXx,
+            colYy=self.colPsfYy,
+            colXy=self.colPsfXy,
+            ellipticityType=self.ellipticityType,
+        )
+
+        calcEDiff = CalcEDiff(colA=calcEMeas, colB=calcEpsf)
+
+        calcSizeResidual = FractionalDifference(
+            actionA=CalcShapeSize(
+                colXx=self.colXx,
+                colYy=self.colYy,
+                colXy=self.colXy,
+                sizeType=self.sizeType,
+            ),
+            actionB=CalcShapeSize(
+                colXx=self.colPsfXx,
+                colYy=self.colPsfYy,
+                colXy=self.colPsfXy,
+                sizeType=self.sizeType,
+            ),
+        )
+
+        # chi-type ellipticity has a shear response of 2, so we need to
+        # divide by 2 so that the rho-stats do not depend on the
+        # ellipticity-type.
+        responsitivity = 2.0 if self.ellipticityType == "chi" else 1.0
```

Comment:
The introduction of `responsitivity` is new and was not present in `analysis_drp`. This ensures that the rho statistics always corresponds to the additive contamination to shear-shear correlation function regardless of the ellipticity type we choose.

Reply:
This is only right in the limit that e is small.  That's probably ok for this purpose, but the distortion responsivity is actually 2-<|e|^2>.

Reply:
The `e` there is intrinsic ellipticity, isn't it? The point source is essentially round, so its intrinsic ellipticity is zero. Any way, this is also set to match what was used with the `reGauss` shapes in the HSC weak lensing analysis code.

Reply:
No, it's the PSF ellipticity.

Reply:
For HSC validation, the `responsitivity` has been set to 2 exactly. Given that the next big validation run will be on HSC, I'd like to keep it this way for consistency.  We can make it more accurate later on, but for LSST, we'd likely be using shear estimator, and not distortion, so it shouldn't matter much any way.

Reply:
So, you want to validate that you get the same (slightly) wrong answer...  OK, I guess.  Wouldn't be my choice though.  I'd calculate the ratio you expect to be different because of this as part of the validation, with comments saying why the HSC values are a little off.